### PR TITLE
Feat/lockdown mode

### PR DESCRIPTION
**Pull request by lorenzowijtman**
Lockdown screen / mode for the app set to show after 31st of january 2022 (user's local time).

It differs form the figma design in the following aspects:
- top fab mini button - commented right now, where should it lead or what should it do?
- show my impact button - commented right now, is this feature currently implemented?

Other changes made:
- Moved the contact form from the ContactFormPage to it's own widget for reuse.
- Added an optional description to the ContentPlaceholder widget
- Added (FAB) svg icons exported from figma design and added flutter_svg package for displaying them

### CODE REVIEW

**Comment by Xazin** [comment]
Do we want flutter_svg dependency when it's not needed? It doesn't render SVGs as the web does, I don't see the reason why we don't just use the PNG versions?

**Comment by lorenzowijtman** [comment]
> Do we want flutter_svg dependency when it's not needed? It doesn't render SVGs as the web does, I don't see the reason why we don't just use the PNG versions?
> 
> All SVGs we have are icons, they should be part of our CollActionIcons.

Fair enough, I see the icon setup now but how do you add icons to it? haven't seen a setup like this before so i have two questions.
- Should the icon be added to the config.json by hand or do you use a specific generator? (didn't see one in pubspec)
- Where do you get the codePoint argument from for the icon in collaction_icons.dart?

**Comment by Xazin** [comment]
> > Do we want flutter_svg dependency when it's not needed? It doesn't render SVGs as the web does, I don't see the reason why we don't just use the PNG versions?
> > 
> > All SVGs we have are icons, they should be part of our CollActionIcons.
> 
> Fair enough, I see the icon setup now but how do you add icons to it? haven't seen a setup like this before so i have two questions.
> - Should the icon be added to the config.json by hand or do you use a specific generator? (didn't see one in pubspec)
> - Where do you get the codePoint argument from for the icon in collaction_icons.dart?

I will look into it as @wizlif did this part and we don't have any documentation on it.

File: lib/presentation/home/home_screen.dart [hunk 1]
```diff
@@ -19,7 +18,15 @@ class _HomePageState extends State<HomePage> {
   @override
   void initState() {
     super.initState();
-    showOnboarding();
+    WidgetsBinding.instance?.addPostFrameCallback((_) => showCaptivePage());
+  }
+
+  Future<void> showCaptivePage() async {
```
Comment (rubenhorn) [inline lib/presentation/home/home_screen.dart:24]: Maybe rename to `checkAndMaybeShowCaptivePage`?

**Review by rubenhorn — APPROVED** [review]

**Comment by wizlif** [comment]
> > > Do we want flutter_svg dependency when it's not needed? It doesn't render SVGs as the web does, I don't see the reason why we don't just use the PNG versions?
> > > 
> > > All SVGs we have are icons, they should be part of our CollActionIcons.
> > 
> > Fair enough, I see the icon setup now but how do you add icons to it? haven't seen a setup like this before so i have two questions.
> > - Should the icon be added to the config.json by hand or do you use a specific generator? (didn't see one in pubspec)
> > - Where do you get the codePoint argument from for the icon in collaction_icons.dart?
> 
> I will look into it as @wizlif did this part and we don't have any documentation on it.

So you can add icons by :-
- Go to [fluttericons.com](fluttericons.com)
- Import/drag and drop the `config.json` to restore existing icons.
- Drag & drop your svgs.
- Select the svg icons & provide their respective names(it automatically uses svg file names).
- Download the new font, it should have `collaction_icons.dart`,`config.json` and `a respective ttf`.
- Copy and replace the respective files above in the flutter project.

**Comment by lorenzowijtman** [comment]
Requested changes have been made but the CI complains about formatting,my IDE doesn't show any additional formatting options or problems though

**Comment by rubenhorn** [comment]
> Requested changes have been made but the CI complains about formatting,my IDE doesn't show any additional formatting options or problems though 

Try running `flutter format .` in the root directory of the repository.

File: lib/presentation/home/captive_screen.dart [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+                  Text(
+                    'ASML used Collaction to host their company-wide Veganuary challenge.',
+                    style: TextStyle(
+                      fontSize: 17,
+                      fontWeight: FontWeight.w300,
+                      color: kPrimaryColor400,
+                    ),
+                    textAlign: TextAlign.center,
+                  ),
+                  // Is this an implemented feature?
```
Comment (rubenhorn) [inline lib/presentation/home/captive_screen.dart:74]: It should link to an online form. Please mark this with `TODO` and link to the homepage for now.

Comment (rubenhorn) [reply lib/presentation/home/captive_screen.dart:74]: Update: Please link to this page: https://www.collaction.org/impact

File: lib/presentation/home/home_screen.dart [hunk 3]
```diff
@@ -19,10 +19,13 @@ class _HomePageState extends State<HomePage> {
...
     super.initState();
-    showOnboarding();
+    WidgetsBinding.instance?.addPostFrameCallback((_) {
+      checkAndMaybeShowOnboarding()
+          .then((_) => checkAndMaybeShowCaptivePage(context.router));
+    });
   }
 
-  Future<void> showOnboarding() async {
+  Future<void> checkAndMaybeShowOnboarding() async {
```
Comment (rubenhorn) [inline lib/presentation/home/home_screen.dart:28]: ❗ **Help requested:**
This method should call `context.router.push(const OnboardingRoute());` instead, but if I change it I always get an exception: `[Root Router] Router can not navigate to onboarding`.

**Review by Xazin — APPROVED** [review]

**Pull request merged**